app/controllers/api/v2/compute_resources_controller.rb

```diff
@@ -237,11 +237,50 @@ def refresh_cache
         end
       end
 
+      api :GET, "/compute_resources/:id/available_virtual_machines/", N_("List available virtual machines for a compute resource")
+      param :id, :identifier, :required => true
+      def available_virtual_machines
+        @available_virtual_machines = @compute_resource.vms.all
+      end
+
+      api :GET, "/compute_resources/:id/available_virtual_machines/:vm_id", N_("Show a virtual machine")
+      param :id, :identifier, :required => true
+      param :vm_id, :identifier, :required => true
+      def show_vm
+        @vm = @compute_resource.find_vm_by_uuid(params[:vm_id])
+        attributes = @vm.attributes.deep_symbolize_keys
+        attributes.delete(:parent)
+        render :json => attributes
+      rescue
```

Comment:
`rescue ActiveRecord::RecordNotFound` ?

Reply:
This is what I wanted to do but it is a bit problematic because when dealing with different compute resources a different error is raised when the vm_id is not found.
For example:
In ovirt the error is:
OvirtSDK4::NotFoundError: HTTP response code is 404. The response body is empty.
In VMware the error is:
ActiveRecord::RecordNotFound

It is actually related to what we discussed above. I think it should be fixed but in a different PR so I will open a Redmine issue for it.